pandas/core/generic.py

```diff
@@ -5783,7 +5784,7 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                   **kwargs)) - 1)
         if freq is None:
             mask = isnull(_values_from_object(self))
-            np.putmask(rs.values, mask, np.nan)
```

Comment:
It doesn't work ;-) I have tried.

Isn't boolean indexation only working for vectors ? Or did you mean rs.loc[mask] (I haven't tried)?

If rs.loc[mask] works, I can replace it.